### PR TITLE
feat(smt): support non-hash data in merkle trees

### PR DESCRIPTION
**Pull request by firelizzard18**
Allow merkle trees to contain data that is not 32-bytes wide.

No functional changes. Verify that nothing breaks.

### CODE REVIEW

**Comment by PaulSnow** [comment]
I am not sure if this is a good idea or not.  

One of the reasons Merkle Trees are hashes only is that the value being added to a Merkle Tree is hidden behind the hash, i.e. I cannot identify a value from its hash except in some cases of dictionary attacks.  If we actually put data into the Merkle Tree, then at the first level, we will be combining the actual data with actual adjacent data, and tracking the actual data in the merkle tree itself.

**Comment by PaulSnow** [comment]
On the other hand, I have no complaint with replacing the Hash [32]byte with a Hash []byte because at some point we might go to a 512 bit hash, and handling []byte for Hashes would make that easier.

**Comment by firelizzard18** [comment]
I was thinking of adding (anchor, chain ID) to the BVN anchor chain. I was also thinking of adding a record to the anchor chain that includes the (Tendermint) block index and time stamp before or after the anchor records for the block.

File: smt/managed/hash.go [hunk 1]
```diff
@@ -10,29 +10,99 @@
...
-// Bytes
-// Return a []byte for the Hash
-func (h Hash) Bytes() []byte {
-	return h[:]
+func (h Hash) Bytes32() [32]byte {
+	if len(h) != 32 {
+		panic("hash is not 32 bytes")
+	}
+	var g [32]byte
+	copy(g[:], h)
```
[the remark this answers is not in the document]
Comment (bunfield) [reply smt/managed/hash.go:28]: NVM, I missed where hash is now a slice instead of fixed size..

**Review by bunfield — APPROVED** [review]
overall it looks good to me.

**Pull request merged**